### PR TITLE
 [WFCORE-4475] jboss-deployment-structure.xml with fails to parse when annotations=true on a sub-deployment module

### PR DESCRIPTION
**Pull request by spyrkob**
JIRA: https://issues.jboss.org/browse/WFCORE-4475

@jmesnil could you review this please? I based this on fix for https://issues.jboss.org/browse/WFCORE-3268, but I'm not sure I check the right resource roots on subdeployments.
I've got a WiP test case for the issue in https://github.com/wildfly/wildfly/compare/master...spyrkob:WFCORE-4475

### CODE REVIEW

**Comment by spyrkob** [comment]
@jamezp sorry, I missed the rebase label notification

File: server/src/main/java/org/jboss/as/server/deployment/annotation/CompositeIndexProcessor.java [hunk 1]
```diff
@@ -82,16 +84,45 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
...
                         indexes.add(indexAttachment);
+                        moduleIndexes.add(indexAttachment);
+                    }
+                }
+                if (!moduleIndexes.isEmpty()) {
+                    additionalAnnotationIndexes.put(moduleIdentifier, new CompositeIndex(moduleIndexes));
+                }
+            } else if (subdeploymentDependencies.containsKey(moduleIdentifier)) {
+                List<ResourceRoot> resourceRoots = subdeploymentDependencies.get(moduleIdentifier).getAttachment(Attachments.RESOURCE_ROOTS);
+                final List<ResourceRoot> allResourceRoots = new ArrayList<ResourceRoot>();
```
Comment (yersan) [inline server/src/main/java/org/jboss/as/server/deployment/annotation/CompositeIndexProcessor.java]: The type in the ArrayList can be inferred by the compiler; I would suggest replacing it by new ArrayList<>()

**Comment by spyrkob** [comment]
Thanks @yersan, I updated the PR with your suggestions

**Comment by jmesnil** [comment]
@spyrkob thanks

**Pull request merged**